File: golang/lib/networks/repartitioner.go
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package networks

import "github.com/kurtosis-tech/kurtosis-libs/golang/lib/core_api_bindings"

type PartitionID string

type Repartitioner struct {
	partitionServices map[PartitionID]*serviceIdSet
	partitionConnections map[PartitionID]map[PartitionID]*core_api_bindings.PartitionConnectionInfo
	defaultConnection *core_api_bindings.PartitionConnectionInfo
}

File: rust/lib/src/networks/repartitioner.rs
use std::collections::{HashMap, HashSet};

use crate::core_api_bindings::api_container_api::PartitionConnectionInfo;

pub struct Repartitioner {
    // TODO convert key to be type alias PartitionID and value to ServiceID
    pub (super) partition_services: HashMap<String, HashSet<String>>,
    // Convert keys of both maps to be PartitionID
    pub (super) partition_connections: HashMap<String, HashMap<String, PartitionConnectionInfo>>,
    pub (super) default_connection: PartitionConnectionInfo,
}
